src/langchain_google_alloydb_pg/async_vectorstore.py

```diff
@@ -482,7 +496,12 @@ async def asimilarity_search(
         **kwargs: Any,
     ) -> List[Document]:
         """Return docs selected by similarity search on query."""
-        embedding = self.embedding_service.embed_query(text=query)
+        embedding = (
+            []
+            if isinstance(self.embedding_service, AlloyDBEmbeddings)
+            else self.embedding_service.embed_query(text=query)
```

Comment:
I was hoping all the embedding service logic could stay here rather than having to call embed_query with the kwargs here and embed_query_inline in another function